server/src/main/java/org/elasticsearch/common/geo/builders/GeometryCollectionBuilder.java

```diff
@@ -183,6 +184,25 @@ public Shape build() {
         //note: ShapeCollection is probably faster than a Multi* geom.
     }
 
+    @Override
+    public Object buildLucene() {
+        List<Object> shapes = new ArrayList<>(this.shapes.size());
+
+        for (ShapeBuilder shape : this.shapes) {
+            Object o = shape.buildLucene();
+            if (o.getClass().isArray()) {
+                shapes.addAll(Arrays.asList(o));
```

Comment:
`shape.buildLucene()` could be an array. For example, a `MultiPolygon`, or a `Polygon` that crosses the dateline will produce a `Polygon[]`.  Or it might produce a point, in which case it should branch to the `else {`

Reply:
> shape.buildLucene() could be an array.

Yes, but the compiler doesn't know that. So I think the compiler is going to treat `o` in 
```
shapes.addAll(Arrays.asList(o));
``` 
as an array with a single element like in 
```
shapes.addAll(Arrays.asList(new Object[]{ o }));
```
 as opposed to an array like in 
```
shapes.addAll(Arrays.asList((Polygon[])o));
```
 Does it makes sense?

Reply:
Pushed fix along w/ an explicit test